pkg/kubelet/kubelet_test.go

```diff
@@ -2793,68 +2817,109 @@ func TestHandlePodResourcesResize(t *testing.T) {
+					ID:        originalPod.UID,
+					Name:      originalPod.Name,
+					Namespace: originalPod.Namespace,
+				}
 
-			updatedPod, err := kubelet.handlePodResourcesResize(newPod, podStatus)
-			require.NoError(t, err)
-			assert.Equal(t, tt.expectedAllocations, updatedPod.Spec.Containers[0].Resources.Requests, "updated pod spec resources")
+				setContainerStatus := func(podStatus *kubecontainer.PodStatus, c *v1.Container, idx int) {
+					podStatus.ContainerStatuses[idx] = &kubecontainer.Status{
+						Name:  c.Name,
+						State: kubecontainer.ContainerStateRunning,
+						Resources: &kubecontainer.ContainerResources{
+							CPURequest:  c.Resources.Requests.Cpu(),
+							CPULimit:    c.Resources.Limits.Cpu(),
+							MemoryLimit: c.Resources.Limits.Memory(),
+						},
+					}
+				}
 
-			alloc, found := kubelet.statusManager.GetContainerResourceAllocation(string(newPod.UID), newPod.Spec.Containers[0].Name)
-			require.True(t, found, "container allocation")
-			assert.Equal(t, tt.expectedAllocations, alloc.Requests, "stored container allocation")
+				if isSidecarContainer {
+					podStatus.ContainerStatuses = make([]*kubecontainer.Status, len(originalPod.Spec.InitContainers))
+					for i, c := range originalPod.Spec.InitContainers {
+						setContainerStatus(podStatus, &c, i)
+					}
+				} else {
+					podStatus.ContainerStatuses = make([]*kubecontainer.Status, len(originalPod.Spec.Containers))
+					for i, c := range originalPod.Spec.Containers {
+						setContainerStatus(podStatus, &c, i)
+					}
+				}
 
-			resizeStatus := kubelet.statusManager.GetPodResizeStatus(newPod.UID)
-			assert.Equal(t, tt.expectedResize, resizeStatus)
+				now := kubelet.clock.Now()
+				// Put the container in backoff so we can confirm backoff is reset.
+				var backoffKey string
+				if isSidecarContainer {
+					backoffKey = kuberuntime.GetStableKey(originalPod, &originalPod.Spec.InitContainers[0])
+				} else {
+					backoffKey = kuberuntime.GetStableKey(originalPod, &originalPod.Spec.Containers[0])
+				}
+				kubelet.backOff.Next(backoffKey, now)
+
+				updatedPod, err := kubelet.handlePodResourcesResize(newPod, podStatus)
+				require.NoError(t, err)
+
+				var updatedPodCtr v1.Container
+				var newPodCtr v1.Container
+				if isSidecarContainer {
+					updatedPodCtr = updatedPod.Spec.InitContainers[0]
+					newPodCtr = newPod.Spec.InitContainers[0]
+				} else {
+					updatedPodCtr = updatedPod.Spec.Containers[0]
+					newPodCtr = newPod.Spec.Containers[0]
+				}
+				assert.Equal(t, tt.expectedAllocations, updatedPodCtr.Resources.Requests, "updated pod spec resources")
 
-			isInBackoff := kubelet.backOff.IsInBackOffSince(backoffKey, now)
-			if tt.expectBackoffReset {
-				assert.False(t, isInBackoff, "container backoff should be reset")
-			} else {
-				assert.True(t, isInBackoff, "container backoff should not be reset")
-			}
-		})
+				alloc, found := kubelet.statusManager.GetContainerResourceAllocation(string(newPod.UID), newPodCtr.Name)
```

Comment:
You don't need `newPodCtr` just to get the name. Name should be immutable, so just use `updatedPodCtr.Name` here instead.